cmd/observe/observe.go

```diff
@@ -210,14 +208,9 @@ programs attached to endpoints and devices. This includes:
 		"to-port", ofilter,
 		"Show only flows with the given destination port (e.g. 8080)"))
 
-	observerCmd.Flags().BoolVarP(
-		&jsonOutput, "json", "j", false, "Use json output",
-	)
-	observerCmd.Flags().BoolVar(
-		&compactOutput, "compact", false, "Use compact output. Automatically used with --follow, unless --json is specified.",
-	)
-	observerCmd.Flags().BoolVar(
-		&dictOutput, "dict", false, "Use dictionary output. Each flow will be shown as KEY:VALUE pairs.",
+	observerCmd.Flags().StringVarP(
+		&output, "output", "o", "compact",
+		"Specify the output format, one of:\n -compact:\tcompact output\n -dict:\teach flow is shown as KEY:VALUE pair\n -json:\tJSON encoding\n",
```

Comment:
Note that there is also a fourth mode, which is actually the current default: The non-compact table mode (selected if `var opts []hubprinter.Option` contains no other mode), which I think should be stay default if `--follow` is absent. We only switch to compact mode when `--follow` is provided.